core/__tests__/models/record/record.ts

```diff
@@ -728,6 +728,38 @@ describe("models/record", () => {
           ).toBeGreaterThan(firstUpdate.getTime());
         });
 
+        test("array property length can grow and shrink", async () => {
```

Comment:
This test would have passed with the old query, but this explicitly confirms that previous array-Property recordProperties are getting removed